armi/reactor/blueprints/tests/test_componentBlueprint.py

```diff
@@ -338,6 +341,48 @@ def test_componentInitializationThoriumNoBurnCustomIsotopics(self):
         for nuc in expectedNuclides:
             self.assertIn(nuc, a[0][0].getNuclides())
 
+    def test_componentBlueprintConstructorWithInvalidMatMods(self):
+        materialBlueprint = r"""
+        nuclide flags:
+            U: {burn: false, xs: true}
+            ZR: {burn: false, xs: true}
+        blocks:
+            fuel: &block_fuel
+                fuel: &component_fuel_fuel
```

Comment:
This test will need to have at least 2 components with different materials to fully test the behavior, i think.